pkg/proxy/proxy_integration_test.go

```diff
@@ -380,11 +357,12 @@ func TestIntegrationGetLoadtestLogs(t *testing.T) {
 
 	for i := 0; i < 5; i++ {
 		//added sleep to wait for kangal controller to create a namespace and start pods
-		testhelper.WaitForResource(5)
+		time.Sleep(4 * time.Second)
```

Comment:
Can this be replaced with the real check of awaited resources with timeout? E.g. check that awaited resources are created/ready every second and stop waiting either when all done, or fail by timeout? This may make tests more stable and in some cases much faster if resources are ready in avg after 2sec and we do not need to wait 2 more.

Reply:
I didn't really focus on proxy tests in this PR, we have other tickets to rewrite them. Current changes were from the previous branch (from non-private repo).

In this case, we run tests in isolation (kind), so I don't really sure that we will need those timeouts there. As I said we will focus on another ticket.